agrona/src/main/java/org/agrona/concurrent/ringbuffer/RingBuffer.java

```diff
@@ -16,11 +16,43 @@
 package org.agrona.concurrent.ringbuffer;
 
 import org.agrona.DirectBuffer;
+import org.agrona.MutableDirectBuffer;
 import org.agrona.concurrent.*;
 
 /**
  * Ring-buffer for the concurrent exchanging of binary encoded messages from producer(s) to consumer(s)
  * in a FIFO manner.
+ *
+ * <p>Example usage of a {@code RingBuffer}:
+ *
+ * <pre>{@code
+ * // Setup
+ * int bufferSize = 1024;
+ * ByteBuffer byteBuffer = ByteBuffer.allocateDirect(bufferSize);
+ * MutableDirectBuffer mutableBuffer = new UnsafeBuffer(byteBuffer);
+ * RingBuffer ringBuffer = new ManyToOneRingBuffer(mutableBuffer);
```

Comment:
I would simplify to a OneToOneRingbuffer because there are no concurrent writers.